include/RAJA/pattern/forall.hpp

```diff
@@ -468,17 +508,22 @@ forall_Icount(ExecutionPolicy&& p,
  *
  ******************************************************************************
  */
-template <typename ExecutionPolicy, typename Res, typename Container, typename LoopBody>
+
+template <typename ExecutionPolicy, typename Res, typename Container, typename... Params>
 RAJA_INLINE concepts::enable_if_t<
     resources::EventProxy<Res>,
     concepts::negate<type_traits::is_indexset_policy<ExecutionPolicy>>,
     concepts::negate<type_traits::is_multi_policy<ExecutionPolicy>>,
     type_traits::is_range<Container>>
-forall(ExecutionPolicy&& p, Res r, Container&& c, LoopBody&& loop_body)
+forall(ExecutionPolicy&& p, Res r, Container&& c, Params&&... params)
 {
   static_assert(type_traits::is_random_access_range<Container>::value,
                 "Container does not model RandomAccessIterator");
 
+  auto f_params = expt::make_forall_param_pack(std::forward<Params>(params)...);
+  auto&& loop_body = expt::get_lambda(std::forward<Params>(params)...);
```

Comment:
Are we sure this is safe? Say the lambda is moved into this call. It seems that forwarding it twice would be an issue. Perhaps if the loop body is grabbed first before the params, it might be harmless to forward twice.